File: Timer/src/Timer.cpp
#include "Timer.hpp"

Timer::Timer()
{
    running = false;
    startTime = std::chrono::time_point<std::chrono::high_resolution_clock>();
    endTime = std::chrono::time_point<std::chrono::high_resolution_clock>();
}

void Timer::start()
{
    startTime = std::chrono::high_resolution_clock::now();
    running = true;
}

void Timer::stop()
{
    endTime = std::chrono::high_resolution_clock::now();
    running = false;
}

double Timer::getDuration()
{
    std::chrono::time_point<std::chrono::high_resolution_clock> endTime;
    if (running)
    {
        endTime = std::chrono::high_resolution_clock::now();
    }
    else
    {
        endTime = this->endTime;
    }
    return (double) std::chrono::duration_cast<std::chrono::milliseconds>(endTime - startTime).count() / 1000.0;
}

File: Python/timer.py
import time

class Timer:
    def __init__(self):
        self.start_time = 0
        self.end_time = 0
        self.running = False

    def start(self):
        self.start_time = time.time()
        self.running = True

    def stop(self):
        self.end_time = time.time()
        self.running = False

    def get_time(self):
        if self.running:
            return time.time() - self.start_time
        else:
            return self.end_time - self.start_time

    def __str__(self):
        minutes, seconds = divmod(self.get_time(), 60)
        return f"Time Taken: {int(minutes):02}:{seconds:06.3f}"
